hazelcast-client/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java

```diff
@@ -554,11 +554,13 @@ public void clear() {
     }
 
     @Override
-    protected void onDestroy() {
-        removeNearCacheInvalidationListener();
-        getContext().getNearCacheManager().destroyNearCache(name);
-
-        super.onDestroy();
+    protected void postDestroy() {
+        try {
+            removeNearCacheInvalidationListener();
```

Comment:
Can't or shouldn't we have the same order here as in the client proxy? Call the super method in the try block and remove the listener in the finally block?